pkg/operator/terminationobserver/termination_metrics.go

```diff
@@ -0,0 +1,65 @@
+package terminationobserver
+
+import (
+	"github.com/blang/semver"
+	"github.com/prometheus/client_golang/prometheus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+type terminationMetrics struct {
+	getApiServersFn func() []*apiServerStatus
+	apiServerMetric *prometheus.GaugeVec
+}
+
+func registerMetrics(getApiServersFn func() []*apiServerStatus) {
+	legacyregistry.MustRegister(&terminationMetrics{
+		getApiServersFn: getApiServersFn,
```

Comment:
this is super ugly to have a callback to an object from a global variable. This also keeps the object around during after tests. Can we change all this logic to avoid this callback to the controller?

Reply:
we can make some in-memory storage that metrics and observer will share, does that sound better?